File: SYP.Web/Modules/ServerTypes/Setting/VendorTypeEditorForm.ts
import { BooleanEditor, DecimalEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface VendorTypeEditorForm {
    Title: StringEditor;
    DiscountType: StringEditor;
    DiscountValue: DecimalEditor;
    Description: StringEditor;
    IsActive: BooleanEditor;
}

export class VendorTypeEditorForm extends PrefixedContext {
    static readonly formKey = 'Setting.VendorTypeEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!VendorTypeEditorForm.init) {
            VendorTypeEditorForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = BooleanEditor;

            initFormType(VendorTypeEditorForm, [
                'Title', w0,
                'DiscountType', w0,
                'DiscountValue', w1,
                'Description', w0,
                'IsActive', w2
            ]);
        }
    }
}

File: SYP.Web/Modules/Setting/VendorType/Editor/VendorTypeEditorForm.cs
namespace SYP.Setting.Forms;

[FormScript("Setting.VendorTypeEditor")]
[BasedOnRow(typeof(VendorTypeRow), CheckNames = true)]
public class VendorTypeEditorForm
{
    public string Title { get; set; }
    public string DiscountType { get; set; }
    public decimal DiscountValue { get; set; }
    public string Description { get; set; }
    public bool IsActive { get; set; }
}